src/neon/intrinsics.rs

```diff
@@ -67,21 +67,214 @@ pub unsafe fn vnegq_s8(a: int8x16_t) -> int8x16_t {
 }
 
 
+/// Extract vector from pair of vectors
+//uint8x16_t vextq_s8 (uint8x16_t a, uint8x16_t b, const int n)
 #[inline]
-fn rotate_(a: u128, b: u128, n: u128) -> u128 {
-    let az = a >> (n * 8);
-    let bz = b << (128 - (n * 8));
-    az | bz
-}
-
-#[inline]
-pub unsafe fn vextq_u8(a: uint8x16_t, b: uint8x16_t, n: u8) -> uint8x16_t {
-    mem::transmute(rotate_(mem::transmute(a), mem::transmute(b), n as u128))
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[rustc_args_required_const(2)]
+pub unsafe fn vextq_u8(a: uint8x16_t, b: uint8x16_t, n: i32) -> uint8x16_t {
+    if n < 0 || n > 15 {
+        unreachable_unchecked();
+    };
+    match n & 0b1111 {
+        0 => simd_shuffle16(a, b, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]),
```

Comment:
for 0, omit the shuffle and just do `0 => a"?

Reply:
Tried that, I'll result in the same assembly output and I found it reads nicer this way :joy:

Reply:
ooh nice!